src/containers/QrScanner.js

```diff
@@ -7,38 +7,65 @@ import QrScanner from '../components/QrScanner'
 import { selectAccount } from '../actions/accounts'
 import { scannedTx } from '../actions/transactions'
 import transaction from '../util/transaction'
+import keccak from '../util/native'
 import store from '../util/store'
 
 var scanning = false
 
+function displayAlert (text) {
+  Alert.alert(text, undefined, [
+    { text: 'OK', onPress: () => { scanning = false } }
+  ])
+}
+
+function findAccountWithAddress (address) {
+  return store.getState().accounts.all.find(account => {
+    return account.address.toLowerCase() === address
+  })
+}
+
 async function onScannedTransaction (data, dispatch) {
   try {
     if (scanning) {
       return
     }
     scanning = true
     let txRequest = JSON.parse(data)
-    let from = txRequest.from.toLowerCase()
-    let account = store.getState().accounts.all.find(account => {
-      return account.address.toLowerCase() === from
-    })
-    if (!account) {
-      Alert.alert('Invalid sender address ' + txRequest.from, undefined, [
-        { text: 'OK', onPress: () => { scanning = false } }
-      ])
-      return
+    switch (txRequest.action) {
+      case 'signTransaction': {
```

Comment:
Syntax is a bit weird, `{}` block is not necessary :)

I prefer early-exiting if statements instead of `switch`, less probability to forget break/return, simpler flow

```js
if (txRequest.action === 'signTransaction') {
   ...
   return
}

if (txRequest.action === 'signTransactionHash') {
  ...
  return
}

displayAlert('')
```